File: java/java-arithmetic/src/main/java/com/leetcode/first/J26.java
package com.leetcode.first;

public class J26 {


    public static void main(String[] args) {

        System.out.println(new J26().removeDuplicates(new int[]{1,1,2,3,3,3,4,4,5}));

    }

    public int removeDuplicates(int[] nums) {

        int preIndex =0;

        int nextIndex =1;

        while (nextIndex<nums.length){
            if (nums[preIndex]==nums[nextIndex]){
                nextIndex++;
            }else{
                int temp=nums[preIndex+1];
                nums[preIndex+1]=nums[nextIndex];
                nums[nextIndex] =temp;
                nextIndex++;
                preIndex++;
            }
        }
//        System.out.println(nums);
        return preIndex+1;

    }

}

File: test-python/leetcode/J26.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        preIndex=0
        nextIndex=1
        for i in range(len(nums)):
            while nextIndex <len(nums):
                if nums[preIndex] == nums[nextIndex]:
                    nextIndex=nextIndex+1
                else:
                    temp=nums[preIndex+1]
                    nums[preIndex+1] = nums[nextIndex]
                    nums[nextIndex]=temp
                    nextIndex=nextIndex+1
                    preIndex=preIndex+1
        return preIndex+1


